prow/gerrit/reporter/reporter_test.go

```diff
@@ -127,13 +150,46 @@ func TestReport(t *testing.T) {
 				},
 				Status: v1.ProwJobStatus{
 					State: v1.SuccessState,
+					URL:   "guber/foo",
+				},
+				Spec: v1.ProwJobSpec{
+					Refs: &v1.Refs{
+						Repo: "foo",
+					},
+					Job: "ci-foo",
 				},
 			},
-			expectReport: true,
-			expectError:  true,
+			expectReport:  true,
+			reportMessage: "1 out of 1 jobs passed!\n\nJob ci-foo finished with success -- URL: guber/foo",
```

Comment:
Not 100% convinced that validating the exact message contents is the most helpful test design.

I suspect using an `expectPresent`, `expectMissing` arrays and doing `strings.Contains("1 out of 1 jobs passed")` `strings.Contains("finished with success")` `!strings.Contains("github.com")` will result in easier to read, less fragile tests.

Reply:
I disagree.

Validating the exact string output is nearly 100% guaranteed to create lots of pointless churn in test cases where any time you change anything (say making `ci-foo` bold, or removing the second `\n` or making this output a bullet list) is going to require changing every test case.

We want to validate that 1 job passed, we are not trying to validate that there are two new lines between `!` and `Job`. Therefore this test should not fail if we change the number of new lines.

There are a bunch of these kinds of tests in the pod-utilities and they are a gigantic pain to work with. I just wind up copying hundreds of lines of boilerplate to change one line because doing anything else is too complicated.

Reply:
done